src/test/java/org/mule/service/http/impl/service/server/grizzly/HttpRequestMultipleAuthenticationMethodsSupportedTestCase.java

```diff
@@ -0,0 +1,151 @@
+
+  private static final int TIMEOUT_MILLIS = 1000;
+  private static final int POLL_DELAY_MILLIS = 200;
+  private static final String DIGEST = "Digest";
+  private static final String BASIC = "Basic";
+  private static final String NONCE =
+      "+Upgraded+v1a574e295ff1f41c52582b82815bb734c5c50331c97c4d301bc97f789c5e9e73ca9564b24cbd898ce5f1c13598999faa2ab013ee5b1597087";
+  private static String chosenAuthMethod = "";
+  private static String TEST_USER = "user";
+  private static String TEST_PASS = "password";
+  private static HttpClient client;
+
+  private HttpClientConfiguration.Builder clientBuilder = new HttpClientConfiguration.Builder().setName("auth-test");
+  private PollingProber pollingProber = new PollingProber(TIMEOUT_MILLIS, POLL_DELAY_MILLIS);
+  private final Reference<HttpResponse> responseReference = new Reference<>();
+
+  public HttpRequestMultipleAuthenticationMethodsSupportedTestCase(String serviceToLoad) {
+    super(serviceToLoad);
+  }
+
+  @Override
+  protected HttpResponse setUpHttpResponse(HttpRequest request) {
+    HttpResponseBuilder responseBuilder = HttpResponse.builder();
+    String authHeader = request.getHeaderValue(AUTHORIZATION);
+    String authType = getRequestAuthType(authHeader);
+    chosenAuthMethod = "";
+
+    if (StringUtils.isEmpty(authHeader)) {
+      responseBuilder.statusCode(UNAUTHORIZED.getStatusCode())
+          .addHeader(WWW_AUTHENTICATE, BASIC)
+          .addHeader(WWW_AUTHENTICATE, NTLM.name())
+          .addHeader(WWW_AUTHENTICATE,
+                     DIGEST + " qop=\"auth\",algorithm=MD5-sess," +
+                         "nonce=\"" + NONCE + "\",charset=utf-8,realm=\"INT\"");
+    } else if (authType.length() > 0) {
+      chosenAuthMethod = authType;
+      responseBuilder.statusCode(OK.getStatusCode())
+          .reasonPhrase(OK.getReasonPhrase());
+    } else {
+      responseBuilder.statusCode(UNAUTHORIZED.getStatusCode());
+    }
+    return responseBuilder.build();
+  }
+
+  public HttpRequestOptions getOptions(HttpAuthenticationBuilder authenticationBuilder) {
+    HttpAuthentication authentication = authenticationBuilder.build();
+    return HttpRequestOptions.builder().authentication(authentication).build();
+  }
+
+  public HttpRequest getRequest() {
+    return HttpRequest.builder().method(GET).uri(getUri()).build();
+  }
+
+  public String getRequestAuthType(String authHeader) {
+    return StringUtils.isEmpty(authHeader) ? "" : authHeader.substring(0, authHeader.indexOf(" "));
+  }
+
+  @Before
+  public void setUpClient() {
+    client = service.getClientFactory().create(clientBuilder.build());
+    client.start();
+  }
+
+  @After
+  public void stopClient() {
+    if (client != null) {
+      client.stop();
+    }
+  }
+
+  @Test
+  public void onMultipleAuthenticationMethodsSupportedLocallySetMethodIsChosenDigest() {
+    sendRequestAssertCorrectAuthMethodChosen(HttpAuthentication.digest(TEST_USER, TEST_PASS), DIGEST);
```

Comment:
You are still repeating the user and pass set up. Use the HttpAuthentication `builder()` method and the builder's `type()` method.